### PR TITLE
create modal

### PR DESCRIPTION
**Pull request by Vadim-Kostetskyi**

### CODE REVIEW

File: src/components/Selector/index.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+export interface CountrySelectorProps {
+  options: SelectOption[];
+  value?: SelectOption;
+  onChange: (
+    newValue: SingleValue<SelectOption>,
+    actionMeta: ActionMeta<SelectOption>,
+  ) => void;
+}
+
+const Selector: FC<CountrySelectorProps> = ({ options, value, onChange }) => {
```
[the remark this answers is not in the document]
Comment (Vadim-Kostetskyi) [reply src/components/Selector/index.tsx:19]: the list can be any, not just cities

File: src/components/HomePage/index.tsx [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
 import CategoryCards from 'components/CategoryCards';
+import { useGetProductsWithImagesQuery } from 'redux/productsApi';
 import Header from 'components/Header';
+import HomePageModal from 'components/HomePageModal';
 import styles from './index.module.scss';
-import { useGetProductsWithImagesQuery } from 'redux/productsApi';
 
 const HomePage = () => {
   const { data } = useGetProductsWithImagesQuery({ page: 0, size: 9 });
   console.log(data);
```
Comment (denred) [inline src/components/HomePage/index.tsx:10]: Please check, I'm not sure that we need this data in the console.

File: src/components/HomePageModal/index.module.scss [hunk 3]
```diff
@@ -0,0 +1,118 @@
+@import 'styles/constants';
+// @import url('~assets/svgs/Check') screen;
+// @import 'assets/svgs/Check';
+// import Check from 'assets/svgs/Check';
```
Comment (denred) [inline src/components/HomePageModal/index.module.scss:4]: please remove if not needed

File: src/components/HomePageModal/index.module.scss [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  bottom: 0;
+  padding: 72px 80px 32px;
+  background-color: $pageBackground;
+}
+
+.modalBox {
+  display: flex;
+  padding-bottom: 75px;
+  margin-bottom: 30px;
+  border-bottom: 1px solid #959595;
```
Comment (denred) [inline src/components/HomePageModal/index.module.scss:28]: Please add this color to the library or use an existing one.

File: src/components/HomePageModal/index.module.scss [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+  input:checked + &::before {
+    background-color: $blackColor;
+    background-image: url('../../assets/svgs/Check.svg');
+  }
+  input:not(:disabled):active + &::before {
+    background-color: $blackColor;
+  }
+  input:disabled + &::before {
+    background-color: #e9ecef;
```
Comment (denred) [inline src/components/HomePageModal/index.module.scss:108]: the same

File: src/components/HomePageModal/index.tsx [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+    (language: string) =>
+      language === selectedLanguage ? styles.focus : styles.languageButton,
+    [],
+  );
+
+  const saveCountryLanguage = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const checkboxChecked = (event.target as HTMLFormElement).check.checked;
+
+    if (!showModal) return null;
```
Comment (denred) [inline src/components/HomePageModal/index.tsx:66]: It seems `if (!showModal) return null;` and the if-else block separately looks clear

File: src/components/copyright/index.module.scss [hunk 7]
```diff
@@ -0,0 +1,5 @@
+.copyright {
+  text-align: center;
+  font-size: 14px;
+  color: #636363;
```
Comment (denred) [inline src/components/copyright/index.module.scss:4]: pls use the existing color or add a new one

**Review by denred — CHANGES_REQUESTED** [review]

File: src/components/HomePage/index.tsx (hunk 2 above)
Comment (Vadim-Kostetskyi) [reply src/components/HomePage/index.tsx:10]: data is not used, so eslint shows an error without a console

Comment (denred) [reply src/components/HomePage/index.tsx:10]: so, this hook is not needed here, pls remove it too

File: src/components/HomePageModal/index.tsx [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+        console.error(error);
+      }
+    } else {
+      setShowAlert(true);
+    }
+  };
+
+  return (
+    <>
+      {showModal ? (
```
Comment (denred) [inline src/components/HomePageModal/index.tsx:87]: It looks like you've already checked the existence of `showModal`

**Review by denred — APPROVED** [review]

**Pull request merged**